packages/msgraph/src/Adaptor.js

```diff
@@ -76,14 +91,205 @@ export function create(resource, data, callback) {
+      handleResponse(response, state, callback)
+    );
+  };
+}
+
+/**
+ * Get a Drive or SharePoint document library. The drive metadata will be written
+ * to state.drives, where it can be used by other adaptor functions.
+ * Pass { id } to get a drive by id or { id, owner } to get default drive for
+ * some parent resource, like a group
+ * @public
+ * @example <caption>Get a drive by ID</caption>
+ * getDrive({ id: "YXzpkoLwR06bxC8tNdg71m" })
+ * @example <caption>Get the default drive for a site</caption>
+ * getDrive({ id: "openfn.sharepoint.com", owner: "sites" })
+ * @param specifier {Object} - A definition of the drive to retrieve
+ *    - id {string} - The ID of the resource or owner.
+ *    - owner {string} - The type of drive owner (e.g. sites, groups).
+ * @param {string} name - The local name of the drive used to write to state.drives, ie, state.drives[name]
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getDrive(specifier, name = 'default', callback = s => s) {
+  return state => {
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedSpecifier, resolvedName] = expandReferences(
+      state,
+      specifier,
+      name
+    );
+
+    const { id, owner = 'drive' } = resolvedSpecifier;
+
+    let resource;
+    if (owner === 'drive') {
+      resource = `drives/${id}`;
+    } else {
+      resource = `${owner}/${id}/drive`;
+    }
+
+    const url = getUrl(resource, apiVersion);
+    const auth = getAuth(accessToken);
+
+    return request(url, { ...auth }).then(response => {
+      state.drives[resolvedName] = response;
+      return callback(state);
+    });
+  };
+}
+
+/**
+ * Get folder metadata or an array of folder contents.
+ * @public
+ * @example <caption>Get a folder by ID</caption>
+ * getFolder('01LUM6XOCKDTZKQC7AVZF2VMHE2I3O6OY3')
+ * @example <caption>Get a folder for a named drive by id</caption>
+ * getFolder("01LUM6XOCKDTZKQC7AVZF2VMHE2I3O6OY3",{ driveName: "mydrive"})
+ * @param {string} pathOrId - A path to a folder or folder id
+ * @param {object} options - (Optional) Query parametes
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getFolder(pathOrId, options, callback = s => s) {
+  return async state => {
+    const defaultOptions = {
+      driveName: 'default', // Named drive in state.drives
+      metadata: false, // If false return folder files if true return folder metadata
+      // $filter: '', // Eg: "file/mimeType eq \'application/vnd.ms-excel\'"
+    };
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedPathOrId, resolvedOptions] = expandReferences(
+      state,
+      pathOrId,
+      options
+    );
+
+    const { driveName, metadata } = { ...defaultOptions, ...resolvedOptions };
+
+    ensureDriveExistsOrFail(state.drives[driveName], 'folder');
+
+    // if (!state.drives[driveName]) {
```

Comment:
We can delete these comments now

Reply:
Done [839975e](https://github.com/OpenFn/adaptors/pull/297/commits/839975e9adb3311d10912eb66c1a419ae55960db)